src/linker/Linker/AssemblyNameHelpers.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace System.Reflection.Runtime.Assemblies
+{
+	internal static partial class AssemblyNameHelpers
+	{
+		internal static string ComputeDisplayName (RuntimeAssemblyName a)
+		{
+			const int PUBLIC_KEY_TOKEN_LEN = 8;
+
+			if (a.Name == string.Empty)
+				throw new FileLoadException ();
+
+			StringBuilder sb = new StringBuilder ();
+			if (a.Name != null) {
+				sb.AppendQuoted (a.Name);
+			}
+
+			if (a.Version != null) {
+				Version canonicalizedVersion = a.Version.CanonicalizeVersion ();
+				if (canonicalizedVersion.Major != ushort.MaxValue) {
```

Comment:
Did you make any changes to these files that we should review?

@marek-safar The fact that this repo uses formatting convention not used by the .NET teams makes source sharing frustrating.

Reply:
I agree and this code should not land under `src/linker/` but under `external/<something>` where it can use any formatting convention and it's clear where it's coming from. I don't know where this code was copied from but it should also copy tests as I'm not comfortable landing hundreds of lines of a parser code without test coverage.

Reply:
This is originally extracted from the runtime reflection stack. The tests for that are the Type.GetType tests in the runtime repo - converting them to something workable is probably a non-trivial amount of work. If we need to service this, the fix should go through the runtime.

Given this code shipped in several major versions of a .NET runtime and the .NET Native compiler, I have pretty high confidence in it even with light (integration-level) test coverage here.

Reply:
@MichalStrehovsky 
 > Did you make any changes to these files that we should review?

Not really. Probably the most interesting change is this - how we parse the assembly version, which is a bit different from how we originally did it for .NET Native, however the code here is how we currently do it in CoreRT: https://github.com/dotnet/corert/blob/9d8bd29a71f68941aaa2c00e9c9a0eed86cafa10/src/System.Private.CoreLib/shared/System/Reflection/AssemblyNameFormatter.cs#L26-L52
That, and the fact that the `ToString` of `NestedTypeName` returns `DeclaringType + "/" Name` instead of `DeclaringType + "+" Name`, because of how Cecil encodes things.